neo/Network/P2P/TaskManager.cs

```diff
@@ -54,16 +56,21 @@ private void OnNewTasks(InvPayload payload)
             HashSet<UInt256> hashes = new HashSet<UInt256>(payload.Hashes);
             hashes.ExceptWith(knownHashes);
             if (payload.Type == InventoryType.Block)
-                session.AvailableTasks.UnionWith(hashes.Where(p => globalTasks.Contains(p)));
-            hashes.ExceptWith(globalTasks);
+                session.AvailableTasks.UnionWith(hashes.Where(p => globalTasks.ContainsKey(p)));
+
+            hashes.ExceptWith(globalTasks.Keys);
```

Comment:
Shouldn't we check `MaxConncurrentTasks` here?

Reply:
hmm, well for blocks it has already added the task to AvailableTasks on line 59. I actually don't think it is that critical to make anything other than the block tasks concurrent. So if this doesn't allow concurrency for anything other than block tasks it is kind of my original intent with the change. However, if we want all tasks to have concurrency then I think it should.

Reply:
I actually would vote to leave it this way and not start additional tasks for non-block tasks.